services/tests/test_repository_service.py

```diff
@@ -1605,24 +1607,22 @@ async def test_fetch_and_update_pull_request_information_torngitexception(
 
     @pytest.mark.asyncio
     async def test_fetch_and_update_pull_request_information_torngitexception_getting_pull(
-        self, dbsession, mocker
+        self, dbsession, mocker, repo
     ):
-        repository = RepositoryFactory.create()
-        dbsession.add(repository)
-        dbsession.flush()
         commit = CommitFactory.create(
             message="",
-            pullid="123",
             totals=None,
             _report_json=None,
-            repository=repository,
+            repository=repo,
         )
         compared_to_commit = CommitFactory.create(
-            repository=repository, branch="master", merged=True
+            repository=repo, branch="master", merged=True
         )
         dbsession.add(commit)
         dbsession.add(compared_to_commit)
         dbsession.flush()
+
+        commit.pullid = "123"
```

Comment:
why is this needed?

Reply:
This was extracted from line 1615. If I leave it in the factory create, it will try to link a pull object and was failing. Since we just needed it to be not null but not actually exist, it was just easier to set it outside the factory call. I don't know why it started failing after I switched to using Django to create the tables 🤷